### PR TITLE
Document how to make an editable install of scipp

### LINKED ISSUE #2915: Editable installs of downstream software depending on scipp?
This may be a problem for developers: Say you develop scipp and scippnexus, and install scippnexus with `pip install -e .` This will install `scipp` via `pip`. However, we may want to instead use our current scipp install (e.g., from `scipp/install`). We could probably force this using PYTHONPATH, but ultimately not being able to have an editable install of scipp causes headaches and surprising issues.

### PR DESCRIPTION
**Pull request by SimonHeybrock**
Fixes #2915.

### CODE REVIEW

File: docs/reference/developer/getting-started.rst [hunk 1]
```diff
@@ -101,6 +101,25 @@ In Python:
 
   import scipp as sc
 
+As an alternative to ``conda develop``, some developers may prefer using ``pip`` to make an editable install.
```
Comment (jl-wynen) [inline docs/reference/developer/getting-started.rst:104]: It is not an alternative. You can use `conda develop src` to make an editable install after installing with cmake.

File: docs/reference/developer/getting-started.rst [hunk 2]
```diff
@@ -101,6 +101,25 @@ In Python:
...
+
+Above we used some of the ``cmake`` presets, but you may also call ``cmake`` without those for more control of the options.
+The crucial part if that the install must be performed into ``src/``.
+You can now use the editable install as usual, i.e., changes to Python files of Scipp are directly visibly when importing Scipp, without need for a new install.
+When making changes to the C++ side of Scipp, you will need to re-run the ``install`` target using ``cmake``, e.g.,
+
+.. code-block:: bash
+
+  cmake --build --preset build
+
```
Comment (nvaytet) [inline docs/reference/developer/getting-started.rst:122]: Do we also want to list the option of just using symbolic links? (which is what I sometimes do?) Or is symbolic links way too old fashioned?

File: docs/reference/developer/getting-started.rst [hunk 3]
```diff
@@ -101,6 +101,25 @@ In Python:
...
+.. code-block:: bash
+
+  cmake --preset base -DCONAN_TBB=ON -DCMAKE_INSTALL_PREFIX=src/
+  cmake --build --preset build
+  pip install -e .
+
+Above we used some of the ``cmake`` presets, but you may also call ``cmake`` without those for more control of the options.
+The crucial part if that the install must be performed into ``src/``.
+You can now use the editable install as usual, i.e., changes to Python files of Scipp are directly visibly when importing Scipp, without need for a new install.
```
Comment (nvaytet) [inline docs/reference/developer/getting-started.rst:116]:
```suggestion
You can now use the editable install as usual, i.e., changes to Python files of Scipp are directly visible when importing Scipp, without the need for a new install.
```

File: docs/reference/developer/getting-started.rst [hunk 4]
```diff
@@ -101,6 +101,25 @@ In Python:
...
   import scipp as sc
 
+As an alternative to ``conda develop``, some developers may prefer using ``pip`` to make an editable install.
+Scipp uses ``scikit-build``, which currently does not fully support this.
+Therefore, we need to call ``cmake`` manually in this case and install into the Python source directory:
+
+.. code-block:: bash
+
+  cmake --preset base -DCONAN_TBB=ON -DCMAKE_INSTALL_PREFIX=src/
+  cmake --build --preset build
```
[the remark this answers is not in the document]
Comment (SimonHeybrock) [reply docs/reference/developer/getting-started.rst:111]: Sounds more complicated and risks breaking stuff?

[the remark this answers is not in the document]
Comment (jl-wynen) [reply docs/reference/developer/getting-started.rst:111]: Maybe. We can keep it as it is and see if it causes problems.

File: docs/reference/developer/getting-started.rst [hunk 5]
```diff
@@ -101,6 +101,34 @@ In Python:
...
+Scipp uses ``scikit-build``, which currently does not fully support this directly.
+Therefore, we need to call ``cmake`` manually in this case and install into the Python source directory, or create symlinks.
+We have configured ``tox`` for this purpose:
+
+.. code-block:: bash
+
+  cmake --preset base -DCONAN_TBB=ON
+  cmake --build --preset build
+  tox -e editable
+  pip install -e .
```
Comment (jl-wynen) [inline docs/reference/developer/getting-started.rst:114]: My previous comment still stands. Instead of this line, you can also use `conda develop src`

Comment (SimonHeybrock) [reply docs/reference/developer/getting-started.rst:114]: But will `pip` know about this?

Comment (jl-wynen) [reply docs/reference/developer/getting-started.rst:114]: No. But why do you need it to?

Comment (jl-wynen) [reply docs/reference/developer/getting-started.rst]: I would never mix the two. Either install all local packages with pip or with conda.

Comment (SimonHeybrock) [reply docs/reference/developer/getting-started.rst]: See update.

**Review by jl-wynen — APPROVED** [review]

File: docs/reference/developer/getting-started.rst [hunk 6]
```diff
@@ -101,6 +101,36 @@ In Python:
...
+Above we used some of the ``cmake`` presets, but you may also call ``cmake`` without those for more control of the options.
+We can also use tox instead of the first two lines:
+
+.. code-block:: bash
+
+  tox -e lib
+  tox -e editable
+  conda develop src
+
+You can now use the editable install as usual, i.e., changes to Python files of Scipp are directly visibly when importing Scipp, without need for a new install.
```
Comment (nvaytet) [inline docs/reference/developer/getting-started.rst:127]:
```suggestion
You can now use the editable install as usual, i.e., changes to Python files of Scipp are directly visible when importing Scipp, without the need for a new install.
```

**Review by nvaytet — APPROVED** [review]

**Pull request merged**